controllers/awscluster_controller.go

```diff
@@ -167,9 +168,8 @@ func (r *AWSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	log = log.WithValues("cluster", klog.KObj(cluster))
 
-	if capiannotations.IsPaused(cluster, awsCluster) {
-		log.Info("AWSCluster or linked Cluster is marked as paused. Won't reconcile")
-		return reconcile.Result{}, nil
+	if isPaused, conditionChanged, err := paused.EnsurePausedCondition(ctx, r.Client, cluster, cluster); err != nil || isPaused || conditionChanged {
+		return ctrl.Result{}, err
 	}
```

Comment:
Maybe we could retain the log line, it is useful in case users are wondering what's going on.
We can talk about the level of logging where this should be printed. What is core CAPI doing in this case?

Reply:
Oh I see it is embedded in the EnsurePausedCondition, so maybe we can keep only that instead?